### ISSUE TITLE
Kops 1.16 spot InstanceGroup stuck in NeedsUpdate

### ISSUE THREAD

**Issue by AlexLast** [kubernetes/kops#8715]
**1. What `kops` version are you running? The command `kops version`, will display
 this information.**

`1.16.0`

**2. What Kubernetes version are you running? `kubectl version` will print the
 version if a cluster is running or provide the Kubernetes version specified as
 a `kops` flag.**

`1.16.7`

**3. What cloud provider are you using?**

`AWS`

**4. What commands did you run?  What is the simplest way to reproduce this issue?**

`kops rolling-update cluster --name cluster --instance-group mixed-spot-ig --yes`

**5. What happened after the commands executed?**

The instance group is rolled successfully and changes are applied, however when running `rolling-update` again the instance group still shows `NeedsUpdate`.

```
NAME			STATUS		NEEDUPDATE	READY	MIN	MAX	NODES
dev			NeedsUpdate	1		0	1	5	1
``` 

**6. What did you expect to happen?**

```
NAME			STATUS		NEEDUPDATE	READY	MIN	MAX	NODES
dev			Ready	        0		1	1	5	1
```

The only thing of note is other instance groups are fine, this bug only seems to affect `spot`/`mixed` instance groups.

**Comment by djablonski-moia** [comment]
Seems like a regression of issue #6796 😢

**Comment by MoShitrit** [comment]
This was fixed in kops 1.16.1+
